subworkflows/nf-core/abundance_differential_filter/main.nf

```diff
@@ -10,111 +10,81 @@ include { CUSTOM_FILTERDIFFERENTIALTABLE      } from '../../../modules/nf-core/c
 
 workflow ABUNDANCE_DIFFERENTIAL_FILTER {
     take:
-    ch_abundance          // [ meta_exp, counts ] with meta keys: method, args_diff
-    ch_transcript_lengths // [ meta_exp, transcript_lengths]
-    ch_control_features   // [ meta_exp, control_features]
-    ch_samplesheet        // [ meta_exp, samplesheet ]
-    ch_contrasts          // [ meta_contrast, contrast_variable, reference, target ]
-    differential_method   // limma, deseq2, propd
-    FC_threshold          // float
-    padj_threshold        // float
+    // Things we may need to iterate
+    ch_input                 // [[meta_input], counts, analysis method,]
```

Comment:
How would you deal with the cases in which, a same method might be run with different parameter options?
Would you convert the current `analysis_method` into a `analysis_meta`?

So, to allow something like this:
```
    withName: "DESEQ2"{
        ext.args = { [
            (meta.args? "${meta.args}" : ''),
            "--p_adjust_method $params.deseq2_p_adjust_method",
            "--alpha $params.deseq2_alpha"
        ].join(' ').trim() }
    }
```

Reply:
For the first question, just copying from Slack for completeness:

> As I mentioned earlier, it's frowned upon to bake meta assumptions into components. Required things should be explicit inputs. Since we need method as part of the iterating logic, I've added it that way. It can then be documented correctly in the meta.yml, for example (I need to do that).
The params are optional, and don't dictate anything about the workflow logic, and we can pass them from the meta via workflow configuration without baking the assumptions into the worklow logic. I'm imagining those params will be part of the meta_input.

Reply:
For the second, I imagine that we'd have duplicate channel content, except for meta content. You'd also have to set publishing paths specific to the meta.